src/player_activity.cpp

```diff
@@ -242,11 +243,11 @@ void player_activity::do_turn( player *p )
             break;
         default:
             // Based on speed, not time
-            if( p->moves <= moves_left ) {
-                moves_left -= p->moves;
+            if( p->moves * season_mult <= moves_left ) {
```

Comment:
> Now that you mention it, though... does C++ still fall victim to that odd behaviour where 8.0 != 8, because 8.0 is technically 7.99999 or 8.000001?

You mean the IEEE floating point standard, which is used basically everywhere? Yes, it does.

And 8.0 is always 8. The only problem comes from code like this:
```C++
double x = 4.0 / 3.0;
double y = x * 3.0;
// This condition might not be true:
if( y == 4.0 ) ...
```